codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -0,0 +1,428 @@
+                HAProxyProtocolVersion.V1, HAProxyCommand.PROXY,
+                protAndFam, parts[2], parts[3], parts[4], parts[5]);
+    }
+
+    /**
+     * Convert ip address bytes to string representation
+     *
+     * @param header     buffer containing ip address bytes
+     * @param addressLen number of bytes to read (4 bytes for IPv4, 16 bytes for IPv6)
+     * @return           string representation of the ip address
+     */
+    private static String ipBytestoString(ByteBuf header, int addressLen) {
+        StringBuilder sb = new StringBuilder();
+        if (addressLen == 4) {
+            sb.append(header.readByte() & 0xff);
+            sb.append('.');
+            sb.append(header.readByte() & 0xff);
+            sb.append('.');
+            sb.append(header.readByte() & 0xff);
+            sb.append('.');
+            sb.append(header.readByte() & 0xff);
+        } else {
+            sb.append(Integer.toHexString(header.readUnsignedShort()));
+            sb.append(':');
+            sb.append(Integer.toHexString(header.readUnsignedShort()));
+            sb.append(':');
+            sb.append(Integer.toHexString(header.readUnsignedShort()));
+            sb.append(':');
+            sb.append(Integer.toHexString(header.readUnsignedShort()));
+            sb.append(':');
+            sb.append(Integer.toHexString(header.readUnsignedShort()));
+            sb.append(':');
+            sb.append(Integer.toHexString(header.readUnsignedShort()));
+            sb.append(':');
+            sb.append(Integer.toHexString(header.readUnsignedShort()));
+            sb.append(':');
+            sb.append(Integer.toHexString(header.readUnsignedShort()));
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Convert port to integer
+     *
+     * @param value                      the port
+     * @return                           port as an integer
+     * @throws HAProxyProtocolException  if port is not a valid integer
+     */
+    private static int portStringToInt(String value) {
+        int port;
+        try {
+            port = Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            throw new HAProxyProtocolException("invalid port: " + value, e);
+        }
+
+        if (port <= 0 || port > 65535) {
+            throw new HAProxyProtocolException("invalid port: " + value + " (expected: 1 ~ 65535)");
+        }
+
+        return port;
+    }
+
+    /**
+     * Validate an address (IPv4, IPv6, Unix Socket)
+     *
+     * @param address                    human-readable address
+     * @param addrFamily                 the {@link AddressFamily} to check the address against
+     * @throws HAProxyProtocolException  if the address is invalid
+     */
+    private static void checkAddress(String address, AddressFamily addrFamily) {
+        if (addrFamily == null) {
+            throw new NullPointerException("addrFamily");
+        }
+
+        switch (addrFamily) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)